web/src/Components/TasksTable.js

```diff
@@ -14,420 +14,446 @@ import TableHead from '@mui/material/TableHead';
-      onClick={() => {
-        triggerSortChange(field);
-      }}
-      sx={{ cursor: 'pointer' }}
-    >
-      <Box sx={{ display: 'flex', flexDirection: 'row' }}>
-        {children}{' '}
-        {sortField.field === field &&
-          (sortField.direction === 'ASC' ? (
-            <KeyboardArrowUpIcon />
-          ) : (
-            <KeyboardArrowDownIcon />
-          ))}
-      </Box>
-    </TableCell>
-  );
+    return (
+        <TableCell
+            onClick={() => {
+                triggerSortChange(field);
+            }}
+            sx={{cursor: 'pointer'}}
+        >
+            <Box sx={{display: 'flex', flexDirection: 'row'}}>
+                {children}{' '}
+                {sortField.field === field &&
+                    (sortField.direction === 'ASC' ? (
+                        <KeyboardArrowUpIcon/>
+                    ) : (
+                        <KeyboardArrowDownIcon/>
+                    ))}
+            </Box>
+        </TableCell>
+    );
 }
 
 const cacheKeyItemsPerPage = 'items_per_page';
 const itemsPerPageList = [10, 25, 50];
 const defaultItemsPerPage = itemsPerPageList[0];
 const getCachedItemsPerPage = () => {
-  const itemsPerPage = Number(localStorage.getItem(cacheKeyItemsPerPage));
-  if (itemsPerPageList.includes(itemsPerPage)) {
-    return itemsPerPage;
-  }
-  return defaultItemsPerPage;
+    const itemsPerPage = Number(localStorage.getItem(cacheKeyItemsPerPage));
+    if (itemsPerPageList.includes(itemsPerPage)) {
+        return itemsPerPage;
+    }
+    return defaultItemsPerPage;
 };
 
 function TasksTable({
-  tasks,
-  sortField,
-  setSortField,
-  relativeDate,
-  onPageChange,
-  page = 1,
-}) {
-  const [itemsPerPage, setItemsPerPage] = useState(getCachedItemsPerPage());
-  const [visibleReasons, setVisibleReasons] = useState([]);
+                        tasks,
+                        sortField,
+                        setSortField,
+                        relativeDate,
+                        onPageChange,
+                        page = 1,
+                    }) {
+    const [itemsPerPage, setItemsPerPage] = useState(getCachedItemsPerPage());
+    const [visibleReasons, setVisibleReasons] = useState([]);
+    const [deployLock, setDeployLock] = useState(false);
 
-  const toggleReason = task => {
-    setVisibleReasons(visibleReasons => {
-      if (visibleReasons.includes(task?.id)) {
-        return [...visibleReasons.filter(id => id !== task?.id)];
-      } else {
-        return [...visibleReasons, task?.id];
-      }
-    });
-  };
+    const checkDeployLock = async () => {
+        const lock = await fetchDeployLock();
+        setDeployLock(lock);
+    };
 
-  const pages = Math.ceil(tasks.length / itemsPerPage);
-  const tasksPaginated = tasks.slice(
-    (page - 1) * itemsPerPage,
-    page * itemsPerPage,
-  );
+    useEffect(() => {
+        checkDeployLock();
+        const intervalId = setInterval(checkDeployLock, 5000); // check every 5 seconds
```

Comment:
I don't think this is the best way to handle the lock status. Is there a better way than calling the API every 5 seconds?

May be fetching the lock intervals for the day? Or smth that doesn't require constant calls to server in background. 

Even if we don't find a better way, I don't think it's worth making constant calls for the lock status. We can fetch the lock status when we need it instad, on demand.

Reply:
Yes, that's a good point. I've replaced it with a simple WebSocket implementation.